documentation/homework/week4/src/stores/InitialStore.js

```diff
@@ -0,0 +1,102 @@
+import { observable, action } from 'mobx'
+
+// Setting our localStorage
+import { locals } from '../utils'
+
+// set the default state of the items Object
+const default_item_state = {
+  id: 0,
+  title: '',
+  deadline: '',
+  done: false,
+  Edit: false,
+}
+
+class Counter {
+  @observable
+  item_state = {...default_item_state}
+
+  @observable
+  item_edit_state = { ...default_item_state }
+
+  // if the localStorage Empty it will be Only ~> []
+  @observable 
+  items = locals.load || [] // Items Holder
+
+  @action // onInput Change
+  onInputChange = (e, field) => (this.item_state = {
+    ...this.item_state,
+    [field]: e.target.value
+  })
+
+  @action
+  addNewItem = () => {
+    const newItem = {
+      ...this.item_state,
+      id: locals.id
```

Comment:
Why is your id generation logic inside your localstorage utils? Should be moved somewhere else, probably inside the store here.